Rename classes in URLParser's characterClassTable

annevk opened the pull request:
#### 81a46f15f08aa96e42a61df330c4bcd821abdb06
<pre>
Rename classes in URLParser&apos;s characterClassTable
<a href="https://bugs.webkit.org/show_bug.cgi?id=283913">https://bugs.webkit.org/show_bug.cgi?id=283913</a>
<a href="https://rdar.apple.com/140789441">rdar://140789441</a>

Reviewed by Alex Christensen.

Rename Default to PathEncode, UserInfo to UserInfoEncode, QueryPercent
to QueryEncode, and SimpleEncode to C0ControlEncode for clarity and to
match the specification.

* Source/WTF/wtf/URLParser.cpp:
(WTF::isInC0ControlEncodeSet):
(WTF::isInFragmentEncodeSet):
(WTF::isInPathEncodeSet):
(WTF::isInUserInfoEncodeSet):
(WTF::shouldPercentEncodeQueryByte):
(WTF::URLParser::parse):
(WTF::URLParser::parseHostAndPort):
(WTF::isInSimpleEncodeSet): Deleted.
(WTF::isInDefaultEncodeSet): Deleted.

Canonical link: <a href="https://commits.webkit.org/287330@main">https://commits.webkit.org/287330@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ffcd6492e2ef7d7960824ce24e9776041663ffc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79075 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58111 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32452 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83714 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30289 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67219 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6381 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61886 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19810 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82142 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51929 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72072 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42190 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49280 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26176 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28653 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/72200 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70396 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26595 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85101 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/78292 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6399 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4426 "Found 2 new test failures: http/wpt/webcodecs/hevc-decoder-annexb.https.any.html http/wpt/webcodecs/hevc-decoder-annexb.https.any.worker.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70128 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6563 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/67944 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69377 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13421 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12208 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/100615 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12228 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6350 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21958 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6309 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9762 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8102 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -185,148 +185,20 @@ static constexpr std::array<uint8_t, 256> characterClassTable {
 template<typename CharacterType> ALWAYS_INLINE static bool isC0Control(CharacterType character) { return character <= 0x1F; }
 template<typename CharacterType> ALWAYS_INLINE static bool isC0ControlOrSpace(CharacterType character) { return character <= 0x20; }
 template<typename CharacterType> ALWAYS_INLINE static bool isTabOrNewline(CharacterType character) { return character <= 0xD && character >= 0x9 && character != 0xB && character != 0xC; }
 template<typename CharacterType> ALWAYS_INLINE static bool isInSimpleEncodeSet(CharacterType character) { return character > 0x7E || isC0Control(character); }
```
Assuming this generally looks okay I will rename this to "isInC0ControlEncodeSet".

achristensen07 commented on Source/WTF/wtf/URLParser.cpp:
```diff
@@ -348,7 +220,9 @@ ALWAYS_INLINE bool URLParser::isForbiddenDomainCodePoint(CharacterType character
 ALWAYS_INLINE static bool shouldPercentEncodeQueryByte(uint8_t byte, const bool& urlIsSpecial)
 {
-    if (characterClassTable[byte] & QueryPercent)
+    if (byte > 0x7F)
```
This is an unnecessary branch in very hot code.  I think it would be better to keep the entire table.

achristensen07 reviewed (commented):
Rename seems fine.  I think the lack of a branch is worth an additional 128 bytes of binary.

annevk committed:
Rename classes in URLParser's characterClassTable
https://bugs.webkit.org/show_bug.cgi?id=283913
rdar://140789441

Reviewed by Alex Christensen.

Rename Default to PathEncode, UserInfo to UserInfoEncode, QueryPercent
to QueryEncode, and SimpleEncode to C0ControlEncode for clarity and to
match the specification.

* Source/WTF/wtf/URLParser.cpp:
(WTF::isInC0ControlEncodeSet):
(WTF::isInFragmentEncodeSet):
(WTF::isInPathEncodeSet):
(WTF::isInUserInfoEncodeSet):
(WTF::shouldPercentEncodeQueryByte):
(WTF::URLParser::parse):
(WTF::URLParser::parseHostAndPort):
(WTF::isInSimpleEncodeSet): Deleted.
(WTF::isInDefaultEncodeSet): Deleted.

Canonical link: https://commits.webkit.org/287330@main
